### PR TITLE
Multi scamps new

### PR DESCRIPTION
**Pull request by zyzzzz-123**
Actually there are many commits ahead of the main branch. 
Sorry about it. It's mainly because it needs more time to do testing experiments than improve some problems. So I always made some new changes before getting a result.

1. Delete useless command line control parameters in main and config.py (only yaml file is left)
2.  Set an upper limit on the number of running processes for multithreading
	(for pure/ubfc/scamps)
3.  Update output format (delete labeled items)
4.  Fix a bug about multi-gpu TSCAN training
5. Changed the way of appending data_path when preprocessing to ensure reproductivity
(Now the results will be the same between DO_PROCESSS: True and False)
6. delete useless yaml files
7. fix "dataloader" name conlict in main_neural_method.py
8. fix using "assert" wrongly problem.

Here are some results:
1.PURE_PURE_UBFC_TSCAN_BASIC   MAE fft ( fft label)  :0.992
2.PURE_PURE_UBFC_DEEPPHYS_BASIC MAE fft ( fft label)  :1.021
3.UBFC_UBFC_PURE_TSCAN.BASIC MAE fft ( fft label)  :5.75
4.UBFC_UBFC_PURE_DEEPPHYS.BASIC MAE fft ( fft label)  : 5.80

### CODE REVIEW

File: configs/PURE_PURE_UBFC_PHYSNET_STANDARDIZED_BASIC.yaml [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    DATA_PATH:   "/data1/toolbox_data/PURE/RawData"       # need to be updated
+    CACHED_PATH: "/data2/rppg_datasets/PreprocessedData/"    # Processed dataset save path, need to be updated
+    EXP_DATA_NAME: ""
+    BEGIN: 0.0
+    END: 0.8
+    PREPROCESS :
+#      DATA_TYPE: [ 'Normalized' ]
+#      LABEL_TYPE: Normalized
+#      DATA_TYPE: [ 'Raw' ]
+#      LABEL_TYPE: Raw
```
Comment (xliucs) [inline configs/PURE_PURE_UBFC_PHYSNET_STANDARDIZED_BASIC.yaml:23]: remove them if you don't need.

File: configs/PURE_PURE_UBFC_PHYSNET_STANDARDIZED_BASIC.yaml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+MODEL:
+  DROP_RATE: 0.2
+  NAME: Physnet
+  PHYSNET:
+    FRAME_NUM: 128
+INFERENCE:
+  BATCH_SIZE: 1
+  MODEL_PATH: "/data1/acsp/Yuzhe_Zhang/Toolbox_master_2/rPPG-Toolbox/PreTrainedModels/PURE_SizeW128_SizeH128_ClipLength128_DataTypeStandardized_LabelTypeStandardized_Large_boxTrue_Large_size1.5_Dyamic_DetFalse_det_len180/PURE_PURE_UBFC_physnet.pth_Epoch6.pth"
+#  MODEL_PATH: "/data1/acsp/Yuzhe_Zhang/Toolbox_master_2/rPPG-Toolbox/PreTrainedModels/PURE_SizeW128_SizeH128_ClipLength128_DataTypeStandardized_LabelTypeStandardized_Large_boxTrue_Large_size1.5_Dyamic_DetFalse_det_len180/PURE_PURE_UBFC_physnet.pth_Epoch9.pth"
+#  "/data1/acsp/Yuzhe_Zhang/Toolbox_master_2/rPPG-Toolbox/PreTrainedModels/PURE_SizeW128_SizeH128_ClipLength128_DataTypeStandardized_LabelTypeStandardized_Large_boxTrue_Large_size1.5_Dyamic_DetFalse_det_len180/PURE_PURE_UBFC_physnet.pth_Epoch11.pth"
```
Comment (xliucs) [inline configs/PURE_PURE_UBFC_PHYSNET_STANDARDIZED_BASIC.yaml:95]: same

File: configs/SCAMPS_SCAMPS_UBFC_PHYSNET_BASIC.yaml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+MODEL:
+  DROP_RATE: 0.2
+  NAME: Physnet
+  PHYSNET:
+    FRAME_NUM: 128
+INFERENCE:
+  BATCH_SIZE: 4
+  MODEL_PATH:   ""
+                    # if TRAIN_OR_TEST:"train_and_test", should be "" or anything, it will use currently trained model.
+                    # if TRAIN_OR_TEST:"only_test", should be "path"
```
Comment (xliucs) [inline configs/SCAMPS_SCAMPS_UBFC_PHYSNET_BASIC.yaml:90]: same as above

File: configs/SCAMPS_SCAMPS_UBFC_PHYSNET_STANDARDIZED_BASIC.yaml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+MODEL:
+  DROP_RATE: 0.2
+  NAME: Physnet
+  PHYSNET:
+    FRAME_NUM: 128
+INFERENCE:
+  BATCH_SIZE: 4
+  MODEL_PATH:   ""
+                    # if TRAIN_OR_TEST:"train_and_test", should be "" or anything, it will use currently trained model.
+                    # if TRAIN_OR_TEST:"only_test", should be "path"
```
Comment (xliucs) [inline configs/SCAMPS_SCAMPS_UBFC_PHYSNET_STANDARDIZED_BASIC.yaml:90]: do you need these?

File: main_neural_method.py [hunk 5]
```diff
@@ -11,6 +11,7 @@
 import argparse
 import glob
 import os
+# os.environ["CUDA_VISIBLE_DEVICES"] = "2"
```
Comment (xliucs) [inline main_neural_method.py:14]: i thought it should be 0?

File: main_neural_method.py [hunk 6]
```diff
@@ -45,22 +46,20 @@ def seed_worker(worker_id):
...
+    # SCAMPS_SCAMPS_UBFC_PHYSNET_STANDARDIZED_BASIC.yaml
+    # SCAMPS_SCAMPS_PURE_DEEPPHYS_BASIC.yaml
+    # SCAMPS_SCAMPS_PURE_TSCAN_BASIC.yaml
+    # PURE_PURE_UBFC_TSCAN_BASIC.yaml
+    # PURE_PURE_UBFC_DEEPPHYS_BASIC.yaml
+    # PURE_PURE_UBFC_PHYSNET_BASIC.yaml
+    # UBFC_UBFC_PURE_TSCAN_BASIC.yaml
+    # UBFC_UBFC_PURE_DEEPPHYS_BASIC.yaml
+    # PURE_PURE_UBFC_PHYSNET_STANDARDIZED_BASIC.yaml
+    # PURE_PURE_UBFC_PHYSNET_NORMARLIZED_BASIC.yaml
```
Comment (xliucs) [inline main_neural_method.py:62]: If you want to keep a list here, maybe make it looks better. Like 
''' Sample Yaml List
    SCAMPS_SCAMPS_UBFC_TSCAN_BASIC.yaml
    SCAMPS_SCAMPS_UBFC_DEEPPHYS_BASIC.yaml
'''

File: metrics/metrics.py [hunk 7]
```diff
@@ -68,78 +68,52 @@ def calculate_metrics(predictions, labels, config):
...
         gt_hr_peak_all.append(gt_hr_peak)
         video_index, GT_HR = read_hr_label(label_dict, index)
-        label_hr.append(GT_HR)
-        if abs(gt_hr_fft - pred_hr_fft) > 10:
+        if abs(gt_hr_fft - pred_hr_fft) > 1:
             print('Video Index: ', video_index)
             print('GT HR fft: ', gt_hr_fft)
             print('Pred HR fft: ', pred_hr_fft)
+            print('GT HR peak: ', gt_hr_peak)
+            print('Pred HR peak: ', pred_hr_peak)
```
Comment (xliucs) [inline metrics/metrics.py:76]: I think it is just for debugging purpose. Don't need to include these if necessary. Also threshold of 1 is too small.

**Review by xliucs — CHANGES_REQUESTED** [review]

File: configs/SCAMPS_SCAMPS_UBFC_PHYSNET_STANDARDIZED_BASIC.yaml (hunk 4 above)
Comment (zyzzzz-123) [reply configs/SCAMPS_SCAMPS_UBFC_PHYSNET_STANDARDIZED_BASIC.yaml:90]: they are comments actually.

**Pull request merged**